aten/src/ATen/native/quantized/cpu/qconcat.cpp

```diff
@@ -0,0 +1,51 @@
+#include <ATen/ATen.h>
+#include <ATen/core/Type.h>
+#include <ATen/core/op_registration/op_registration.h>
+#include <ATen/native/TensorIterator.h>
+#include <ATen/native/cpu/Loops.h>
+
+#include <algorithm>
+#include <vector>
+
+namespace at { namespace native {
+namespace {
+template <bool ReLUFused = false>
+class QCat final : public torch::OperatorKernel {
+ public:
+  Tensor operator()(const std::vector<Tensor>& qxs, int64_t axis,
+                    double scale, int64_t zero_point) {
+    const auto x_dtype = qxs[0].scalar_type();
+    std::vector<Tensor> xs;
+    xs.reserve(qxs.size());
+    for (const auto& qx: qxs) {
+      TORCH_CHECK(x_dtype == qx.scalar_type(),
+                   "All dtypes must be the same.");
+      xs.push_back(qx.dequantize());
+    }
```

Comment:
Also, cat should only work for per tensor quantization.

Reply:
Can you, please, elaborate?

Reply:
I guess it doesn't make sense to concatenate tensors that are per channel quantized?

Also, what is the common use case? Is it common to have a list of tensors that have the same scale and zero_point?

Reply:
Will put the checks